### PR TITLE
Add fields validators, queries, commands

### PR DESCRIPTION
**Pull request by Warchant**
Signed-off-by: Bogdan Vaneev <warchantua@gmail.com>

- [x] add new commands. Now this lib supports ALL v4 commands
- [x] add new queries. now this lib supports ALL v4 queries
- [x] remove Signer, as it duplicates functionality
- [x] increase test coverage
- [x] change return type of IrohaAPI.transaction and IrohaAPI.query from Subject to Observable.

### CODE REVIEW

File: src/main/java/jp/co/soramitsu/iroha/java/FieldValidator.java [hunk 1]
```diff
@@ -49,56 +94,108 @@ public void checkQuorum(int quorum) {
...
+          assetId);
+    }
+
+    try {
+      this.checkAssetName(t[0]);
+      this.checkDomain(t[1]);
+    } catch (ValidationException e) {
+      throw new ValidationException(
+          ASSET_ID,
+          "Valid format is account@domain, got '%s'. Details: '%s'.",
```
Comment (rogachevsn55) [inline src/main/java/jp/co/soramitsu/iroha/java/FieldValidator.java]:
```suggestion
          "Valid format is asset#domain, got '%s'. Details: '%s'.",
```

File: src/main/java/jp/co/soramitsu/iroha/java/FieldValidator.java [hunk 2]
```diff
@@ -49,56 +94,108 @@ public void checkQuorum(int quorum) {
...
+          "Invalid key. Expected '%s', got '%s'",
+          accountDetailsKeyPattern.pattern(),
+          key
+      );
+    }
   }
 
   public void checkAccountDetailsValue(@NonNull String value) {
-    // TODO
+    int len = 4096;
```
Comment (rogachevsn55) [inline src/main/java/jp/co/soramitsu/iroha/java/FieldValidator.java]: extract this value to constant

File: src/main/java/jp/co/soramitsu/iroha/java/TransactionBuilder.java [hunk 3]
```diff
@@ -140,6 +143,23 @@ public TransactionBuilder createAccount(
...
 
+  public TransactionBuilder createAccount(
+      String accountId,
+      PublicKey publicKey
+  ) {
+    if (nonNull(this.validator)) {
+      this.validator.checkAccountId(accountId);
+    }
+
+    val t = accountId.split("@");
```
Comment (rogachevsn55) [inline src/main/java/jp/co/soramitsu/iroha/java/TransactionBuilder.java]: extract delimiter to a constants from here and FormValidator

File: src/main/java/jp/co/soramitsu/iroha/java/FieldValidator.java (hunk 2 above)
Comment (Warchant) [reply src/main/java/jp/co/soramitsu/iroha/java/FieldValidator.java]: fixed

File: src/main/java/jp/co/soramitsu/iroha/java/TransactionBuilder.java (hunk 3 above)
Comment (Warchant) [reply src/main/java/jp/co/soramitsu/iroha/java/TransactionBuilder.java]: fixed

**Pull request merged**